### PR TITLE
feat: detect firecracker host support in config init and doctor

### PR DESCRIPTION
**Pull request by lox**
## Summary

This adds shared Firecracker host support detection and uses it in `cleanroom config init` and `cleanroom doctor`.

It does three main things:
- adds a shared Firecracker host probe for helper, snapshot runtime, and ZFS dataset-root detection
- makes `cleanroom config init` pick `zfs` when an existing usable Cleanroom dataset root is detected, otherwise fall back to `file` or disabled snapshots with a warning
- adds a Firecracker support-tier summary to `cleanroom doctor`, so the CLI can distinguish supported ZFS-backed hosts from degraded file-backed hosts and missing bootstrap state

The docs now frame Linux Firecracker as machine bootstrap plus unprivileged user config, with ZFS-backed Firecracker as the supported layered-cache path.

## Testing

- `go test ./...`
- read-only SSM checks on prod host `i-0b35e6770b45a7db1`

## Prod host note

The prod host test surfaced one useful edge case: its usable ZFS root is a dedicated `cleanroom` pool root, not only `*/cleanroom`. This PR includes that fix, so the host detector now accepts both shapes.

The same host is currently configured for file-backed Firecracker snapshots, so under this PR it would be classified as degraded rather than unsupported. It is already machine-bootstrap capable of the ZFS-backed path.

### CODE REVIEW

File: internal/backend/firecracker/host_support.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+		result.RuntimeMessage = fmt.Sprintf("firecracker host runtime bootstrap is only available on linux (current host: %s)", runtime.GOOS)
+		result.SnapshotMessage = result.RuntimeMessage
+		result.ZFSMessage = result.RuntimeMessage
+		return result
+	}
+
+	missingCommands := make([]string, 0, len(result.RequiredCommands))
+	for _, command := range result.RequiredCommands {
+		if _, err := hostSupportLookPath(command); err != nil {
+			missingCommands = append(missingCommands, command)
```
[the remark this answers is not in the document]
Comment (lox) [reply internal/backend/firecracker/host_support.go:56]: Fixed in fb730603: required host commands now resolve with the same sbin fallback paths the helper runtime expects, so hosts with tools outside an unprivileged PATH are detected correctly.

File: internal/backend/firecracker/host_support.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+func validateZFSDatasetRoot(ctx context.Context, zfsPath, dataset string) error {
+	dataset = strings.TrimSpace(dataset)
+	if dataset == "" {
+		return fmt.Errorf("zfs dataset root is empty")
+	}
+	if !isCleanroomZFSDatasetRoot(dataset) {
+		return fmt.Errorf("zfs dataset root %q must be cleanroom or */cleanroom", dataset)
+	}
+	out, err := hostSupportCommandOutput(ctx, zfsPath, "list", "-H", "-d", "0", "-o", "name", dataset)
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (lox) [reply internal/backend/firecracker/host_support.go]: Fixed in aca5ecd: configured ZFS dataset validation now goes through the privileged helper-backed `zfs list` path in both host support detection and Firecracker doctor, with regression coverage for helper-only access.

**Comment by lox** [comment]
@codex review
